src/java.base/share/classes/java/security/cert/PolicyQualifierInfo.java

```diff
@@ -89,6 +91,28 @@
     private byte [] mData;
     private String pqiString;
 
+    /**
+     * Creates from material.
+     * @param mId id
+     * @param mData data
+     */
+    public PolicyQualifierInfo(String mId, byte[] mData) {
+        this.mId = mId;
+        this.mData = mData;
+    }
+
+    private void encodeThis() throws IOException {
+        if (mEncoded != null) {
+            return;
+        }
+        DerOutputStream out = new DerOutputStream();
+        DerOutputStream ins = new DerOutputStream();
+        ins.putOID(ObjectIdentifier.of(mId));
+        ins.write(mData);
+        out.write(DerValue.tag_Sequence, ins);
+        mEncoded = out.toByteArray();
+    }
+
```

Comment:
This seem irrelevant to the RFE? It seems that existing impl only supports parsing? Are we really using this new constructor somewhere in JDK?

Reply:
I uses the constructor in a test. Yes it should not be included.